File: TOMIMSV4/Entity/Mantenimientos/Acuerdos_Comerciales/Trans_AcuerdosComerciales_Enc/clsBeTrans_acuerdoscomerciales_enc.vb
Public Class clsBeTrans_acuerdoscomerciales_enc
    Implements ICloneable

    Public Property IdAcuerdoEnc() As Integer = 0
    Public Property IdCliente() As Integer = 0
    Public Property Codigo_acuerdo() As Integer = 0
    Public Property Descripcion() As String = ""
    Public Property Tipo_cobro() As String = ""
    Public Property Cod_moneda() As Integer = 0
    Public Property Moneda() As String = ""
    Public Property Estado() As Boolean = False
    Public Property user_agr As String = ""
    Public Property fec_agr As DateTime = Now
    Public Property user_mod As String = ""
    Public Property fec_mod As DateTime = Now
    Public Property fec_erp As DateTime = Now
    Sub New()
        '#EJC: Add your constructor here... 
    End Sub

    Public Function Clone() As Object Implements System.ICloneable.Clone
        Return MyBase.MemberwiseClone()
    End Function

End Class

File: WMS.EntityCore/Acuerdos/clsBeTrans_acuerdoscomerciales_enc.cs
namespace WMS.EntityCore.AcuerdosComerciales
{
    public class clsBeTrans_acuerdoscomerciales_enc : ICloneable
    {
        public int IdAcuerdoEnc { get; set; } = 0;
        public int IdCliente { get; set; } = 0;
        public int? Codigo_acuerdo { get; set; } = null;
        public string Descripcion { get; set; } = "";
        public string Tipo_cobro { get; set; } = "";
        public int? Cod_moneda { get; set; } = null;
        public string Moneda { get; set; } = "";
        public bool? Estado { get; set; } = true;
        public string User_agr { get; set; } = "";
        public DateTime? Fec_agr { get; set; } = DateTime.Now;
        public string User_mod { get; set; } = "";
        public DateTime? Fec_mod { get; set; } = DateTime.Now;
        public DateTime? Fec_erp { get; set; } = null;

        /// <summary>
        /// Campo adicional para control de envío a ERP
        /// </summary>
        public bool Enviado_A_ERP { get; set; } = false;

        /// <summary>
        /// Campo adicional para auditoría interna
        /// </summary>
        public bool Auditado { get; set; } = false;

        public clsBeTrans_acuerdoscomerciales_enc()
        {
        }

        public clsBeTrans_acuerdoscomerciales_enc(
            int idAcuerdoEnc,
            int idCliente,
            string descripcion,
            string moneda,
            string user_agr,
            DateTime fec_agr)
        {
            IdAcuerdoEnc = idAcuerdoEnc;
            IdCliente = idCliente;
            Descripcion = descripcion ?? "";
            Moneda = moneda ?? "";
            User_agr = user_agr ?? "";
            Fec_agr = fec_agr;
        }

        public object Clone()
        {
            return this.MemberwiseClone();
        }
    }
}